Mathlib/Data/Finset/Grade.lean

```diff
@@ -0,0 +1,158 @@
+  ordConnected_range_coe.apply_covby_apply_iff ⟨⟨_, coe_injective⟩, coe_subset⟩
+
+alias ⟨_, _root_.wcovby.finset_val⟩ := val_wcovby_val
+alias ⟨_, _root_.Covby.finset_val⟩ := val_covby_val
+alias ⟨_, _root_.wcovby.finset_coe⟩ := coe_wcovby_coe
+alias ⟨_, _root_.Covby.finset_coe⟩ := coe_covby_coe
+
+@[simp] lemma covby_cons (ha : a ∉ s) : s ⋖ s.cons a ha := by simp [← val_covby_val]
+
+lemma _root_.Covby.exists_finset_cons (h : s ⋖ t) : ∃ a, ∃ ha : a ∉ s, t = s.cons a ha :=
+  let ⟨a, ha, hst⟩ := ssubset_iff_exists_cons_subset.1 h.lt
+  ⟨a, ha, hst.eq_of_not_ssuperset $ h.2 $ ssubset_cons _⟩
+
+lemma covby_iff_exists_cons : s ⋖ t ↔ ∃ a, ∃ ha : a ∉ s, t = s.cons a ha :=
+  ⟨Covby.exists_finset_cons, by rintro ⟨a, ha, rfl⟩; exact covby_cons _⟩
+
+lemma _root_.Covby.card_finset (h : s ⋖ t) : s.card ⋖ t.card := (val_covby_val.2 h).card_multiset
+
+section DecidableEq
+variable [DecidableEq α]
+
+@[simp] lemma wcovby_insert (s : Finset α) (a : α) : s ⩿ insert a s := by simp [← coe_wcovby_coe]
+@[simp] lemma erase_wcovby (s : Finset α) (a : α) : s.erase a ⩿ s := by simp [← coe_wcovby_coe]
+
+lemma covby_insert (ha : a ∉ s) : s ⋖ insert a s :=
+  (wcovby_insert _ _).covby_of_lt $ ssubset_insert ha
+
+@[simp] lemma erase_covby (ha : a ∈ s) : s.erase a ⋖ s := ⟨erase_ssubset ha, (erase_wcovby _ _).2⟩
+
+lemma _root_.Covby.exists_finset_insert (h : s ⋖ t) : ∃ a ∉ s, t = insert a s := by
+  simpa using h.exists_finset_cons
+
+lemma _root_.Covby.exists_finset_erase (h : s ⋖ t) : ∃ a ∈ t, s = t.erase a := by
+  simpa only [← coe_inj, coe_erase] using h.finset_coe.exists_set_sdiff_singleton
+
+lemma covby_iff_exists_insert : s ⋖ t ↔ ∃ a ∉ s, t = insert a s := by
+  simp only [← coe_covby_coe, Set.covby_iff_exists_insert, ← coe_inj, coe_insert, mem_coe]
+
+lemma covby_iff_card_sdiff_eq_one : t ⋖ s ↔ t ⊆ s ∧ (s \ t).card = 1 := by
+  rw [covby_iff_exists_insert]
+  constructor
+  · rintro ⟨a, ha, rfl⟩
+    simp [*]
+  · simp_rw [card_eq_one]
+    rintro ⟨hts, a, ha⟩
+    refine ⟨a, (mem_sdiff.1 $ superset_of_eq ha $ mem_singleton_self _).2, ?_⟩
+    rw [insert_eq, ← ha, sdiff_union_of_subset hts]
+
+lemma covby_iff_exists_erase : s ⋖ t ↔ ∃ a ∈ t, s = t.erase a := by
+  simp only [← coe_covby_coe, Set.covby_iff_exists_sdiff_singleton, ← coe_inj, coe_erase, mem_coe]
+
+end DecidableEq
+
+@[simp] lemma isAtom_singleton (a : α) : IsAtom ({a} : Finset α) :=
+  ⟨singleton_ne_empty a, fun _ ↦ eq_empty_of_ssubset_singleton⟩
+
+protected lemma isAtom_iff : IsAtom s ↔ ∃ a, s = {a} :=
+  bot_covby_iff.symm.trans $ covby_iff_exists_cons.trans $ by simp
+
+section Fintype
+variable [Fintype α] [DecidableEq α]
+
+lemma isCoatom_compl_singleton (a : α) : IsCoatom ({a}ᶜ : Finset α) := (isAtom_singleton a).compl
+
+protected lemma isCoatom_iff : IsCoatom s ↔ ∃ a, s = {a}ᶜ := by
+  simp_rw [← isAtom_compl, Finset.isAtom_iff, compl_eq_iff_isCompl, eq_compl_iff_isCompl]
+
+end Fintype
+
+instance instGradeMinOrder_multiset : GradeMinOrder (Multiset α) (Finset α) where
```

Comment:
This should be in the module docstring, and perhaps is deserving of a sentence or two of explanation here too.

Reply:
I did the other way around, adding a docstring but not mentioning it in the module doc since I believe it's not mathematically very meaningful.